### ISSUE TITLE
💅 noFloatingPromises false positives in 2.5.11

### ISSUE THREAD

**Issue by jakeleventhal** [biomejs/biome#11528]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.11
  Color support:                false

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:                npm
  JS_RUNTIME_VERSION:           v24.5.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.4.0

Biome Configuration:
  Status:                       Loaded successfully.
  Linter enabled:               true
```

</details>

### Rule name

`lint/nursery/noFloatingPromises`

### Playground link

https://github.com/jakeleventhal/biome-no-floating-promises-repro

Minimal reproduction repository with two cases:

1. **TanStack Query `invalidateQueries`** — fire-and-forget usage in mutation callbacks (`src/invalidate-queries.ts`). TanStack Query documents awaiting as optional; Biome 2.5.11 flags the call as a floating promise. Biome 2.5.10 does not.

2. **`google-auth-library` `OAuth2Client.revokeToken`** — single `await` on a `GaxiosPromise` return type (`src/gaxios-promise.ts`). Biome 2.5.11 flags the awaited expression as a floating promise and suggests `await await`. Biome 2.5.10 does not.

```bash
bun install
bun run check
```

### Expected result

- `invalidateQueries` without `await`/`void` should not be reported when callers intentionally fire-and-forget (or the rule should not treat this as an error for well-typed TanStack Query usage).
- `await client.revokeToken(token)` should not be reported; a single `await` is correct for `GaxiosPromise<T>` (`Promise<GaxiosResponse<T>>`).

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
